python/ray/experimental/workflow/tests/test_object_deref.py

```diff
@@ -0,0 +1,55 @@
+import ray
+from ray.experimental import workflow
+
+
+@ray.remote
+def nested_ref():
+    return ray.put(42)
+
+
+@workflow.step
+def nested_workflow(n):
+    if n <= 0:
+        return "nested"
+    else:
+        return nested_workflow.step(n - 1)
+
+
+@workflow.step
+def deref_check(u, v, w, x, y, z):
+    try:
+        return (u == 42 and ray.get(v) == 42 and ray.get(ray.get(w[0])) == 42
+                and x == "nested" and y[0] == "nested"
+                and z[0]["output"] == "nested")
+    except Exception:
+        return False
+
+
+@workflow.step
```

Comment:
Could we also test that `Workflow` is only serializable when passed as an arg or return value? For example, the following should fail to serialize:

- ray.put(workflow)
- closure capture of a workflow variable in a remote function
- returning workflow object in a data structure from a remote function